### PR TITLE
lantiq: clean up package names

### PR DESCRIPTION
**Pull request by dhewg**
This cleans up some directory/PKGNAME mismatches and also prepares for upcoming versions, like VR11, which we'll hopefully seen soon.

### CODE REVIEW

**Comment by xdarklight** [comment]
This looks sane, the reason behind it makes sense and so this gets my:
`Acked-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>`

File: package/network/config/ltq-adsl-app/Makefile [hunk 1]
```diff
@@ -36,6 +36,7 @@ define Package/ltq-adsl-app
   TITLE:=Lantiq DSL userland tool
   URL:=http://www.lantiq.com/
   DEPENDS:=@(TARGET_lantiq_xway||TARGET_lantiq_xway_legacy||TARGET_lantiq_ase) +libpthread +ltq-dsl-base +libubox +libubus
+  PROVIDES:=ltq-dsl-app
```
Comment (Ansuel) [inline package/network/config/ltq-adsl-app/Makefile:39]: just for reference the tool export the same stuff or we can encounter some compatibility problem?

Comment (dhewg) [reply package/network/config/ltq-adsl-app/Makefile:39]: There may be `ltq-vdsl-vr11-app` in the future, and with that all three are from the same codebase, just different versions supporting different hw generations (think branches).

The primary use is for consumers for the ubus stats those three provide, at this point that's `luci-mod-status`, `prometheus-node-exporter-lua` and `collectd`. Luci detects it at runtime, but one contains `DEPENDS:=@(PACKAGE_ltq-adsl-app||PACKAGE_ltq-vdsl-app)` and the other `@TARGET_lantiq_xrx200`. With ipq40xx/ltq-vdsl-vr11-app on top that'll get a bit ugly, so this is for them switching to just `ltq-dsl-app`.

I was using `prometheus-node-exporter-lua` on lantiq/vr9 and now I'm using it on ipq40xx/vr11, and as far as I can tell both provide the very same stats for my dsl line, so I don't expect any compatibility problems.

**Comment by dhewg** [comment]
^ the package PR. Even without vr11, I think this is worth having ;)

**Comment by hauke** [comment]
Thank you for the patches, I applied them to master in 5d53b8e9f91c7e434861c30bb4d62ffe0c5b121e, 07536cff513f38ea167c3b1d870d27f1d71a292d, 33e2115fe449d1b9e97ce8c5be96f580442fd3d9 and 1afd0fefd22992b808e98fcf726a8e4d3af91604.

**Pull request closed without merging**